source/common/upstream/upstream_impl.h

```diff
@@ -494,6 +493,56 @@ class ClusterImplBase : public Cluster, protected Logger::Loggable<Logger::Id::u
   uint64_t pending_initialize_health_checks_{};
 };
 
+typedef std::unique_ptr<HostVector> HostListPtr;
+typedef std::unordered_map<envoy::api::v2::core::Locality, uint32_t, LocalityHash, LocalityEqualTo>
+    LocalityWeightsMap;
+typedef std::vector<std::pair<HostListPtr, LocalityWeightsMap>> PriorityState;
+
+/**
+ * Manages PriorityState of a cluster. PriorityState is a per-priority binding of a set of hosts
+ * with its corresponding locality weight map. This is not used at runtime, but useful to store
```

Comment:
Sorry, can you clarify what you mean by this not being used at runtime?

Reply:
Probably the wording is not right. I meant to say: it is configuration related only, not in data path.